[DW-159] Adding website document types to local search

giacomolm opened the pull request.

wojtek-oledzki commented on repository-data/application/src/main/resources/hcm-content/content/documents/administration/website/search-tab-labels.yaml:
```diff
@@ -0,0 +1,70 @@
+---
```
just add new labels to existing "administration/searchtab-labels"

jacsamell commented on site/src/main/java/uk/nhs/digital/common/enums/SearchArea.java:
```diff
@@ -4,10 +4,12 @@
     DATA,
     SERVICES,
-    NEWS;
+    NEWS,
```
Do we really want tabs for General and Hubs? We have previously decided with Owain to only have the tabs for All, Data and Information, Services and News and not to add any new ones.

giacomolm replied:
good point, this is part of the next US https://jira.digital.nhs.uk/browse/DW-223. Does it make sense to push that fix on a different branch/PR ?

jacsamell replied:
Would probably be easier to do it now as you'll end up changing exactly the same code here but it's up to you, I don't really mind 🙂

jacsamell commented on site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java:
```diff
@@ -271,6 +282,35 @@ private HstQuery constructQuery(HstQueryBuilder queryBuilder, Constraint searchS
     /**
      * Publication System content types that should be included in search
      */
+    private void deriveTypes(final HstQueryBuilder query, final HstRequest request) {
```
Just FYI I think I'm going to remove this and replace it with one single list of all the doctypes (RPS + NIL + DW) and then let the folder(/s) be the single source of filtering for the tabs. Happy to merge like this and I'll fix it later. Have discussed with @wojtek-oledzki as I know he told you the opposite originally 😉

giacomolm replied:
I am pushing the changes for DW-223 directly in this pull request, I think it's easier to sort it out here. I will update this PR shortly

jacsamell replied:
👍

giacomolm commented:
This is the list of files changed according to DW-223
	deleted:    repository-data/application/src/main/resources/hcm-content/content/documents/administration/website/search-tab-labels.yaml
	modified:   repository-data/webfiles/src/main/resources/site/freemarker/common/searchtabs.ftl
	modified:   site/src/main/java/uk/nhs/digital/common/components/SearchComponent.java
	modified:   site/src/main/java/uk/nhs/digital/common/enums/SearchArea.java
